app/src/main/java/de/westnordost/streetcomplete/location/FineLocationManager.kt

```diff
@@ -41,8 +41,20 @@ class FineLocationManager(context: Context, locationUpdateCallback: (Location) -
         }
     }
 
+    // Both signals are refreshed regardless of whether the device has both providers, because
+    // they are both canceled in removeUpdates and both checked in the locationListener
+    private fun refreshCancellationSignals() {
+        if (gpsCancellationSignal.isCanceled) {
+            gpsCancellationSignal = CancellationSignal()
+        }
+        if (networkCancellationSignal.isCanceled) {
+            networkCancellationSignal = CancellationSignal()
+        }
+    }
+
     @RequiresPermission(ACCESS_FINE_LOCATION)
     fun requestUpdates(minTime: Long, minDistance: Float) {
+        refreshCancellationSignals()
         if (deviceHasGPS)
             locationManager.requestLocationUpdates(GPS_PROVIDER, minTime, minDistance, locationListener, Looper.getMainLooper())
```

Comment:
I suppose it's a little awkward to refresh the signals here, given that the `requestLocationUpdates` callback doesn't actually *use* the signal, except to check its status in the callback.